server/cli/src/test_connection.rs

```diff
@@ -0,0 +1,499 @@
+            };
+            gui_tx.send(gui_state.clone()).await.unwrap();
+        }
+    }
+
+    info!("All tests completed");
+}
+
+struct TestData {
+    server_config: Option<service::settings::Settings>,
+    sync_api_v5: Option<SyncApiV5>,
+    args: Args,
+}
+
+#[async_trait]
+trait Test {
+    async fn run(&self, test_data: &mut TestData) -> Result<String>;
+}
+
+struct ConfigTest;
+
+#[async_trait]
+impl Test for ConfigTest {
+    async fn run(&self, test_data: &mut TestData) -> Result<String> {
+        test_data.server_config = Some(
+            configuration::get_configuration()
+                .map_err(|err| anyhow!("Failed to load config: {:?}", err))?,
+        );
+        Ok("Successfully loaded configuration".to_string())
+    }
+}
+
+struct PingTest;
+
+#[async_trait]
+impl Test for PingTest {
+    async fn run(&self, test_data: &mut TestData) -> Result<String> {
+        let config = test_data
+            .server_config
+            .as_ref()
+            .ok_or(anyhow!("No config loaded".to_string()))?;
+
+        let url = get_url(config)?;
+
+        info!("Pinging server at: {}", url.to_string());
+
+        let response = reqwest::get(url)
+            .await
+            .map_err(|err| anyhow!("Failed to get response: {:?}", err))?;
+
+        if response.status().is_success() {
+            Ok("Successfully pinged server".to_string())
+        } else {
+            Err(anyhow!("Failed to ping server: {:?}", response))
+        }
+    }
+}
+
+struct DatabaseTest;
+
+#[async_trait]
+impl Test for DatabaseTest {
+    async fn run(&self, test_data: &mut TestData) -> Result<String> {
+        let config = test_data
+            .server_config
+            .as_ref()
+            .ok_or(anyhow!("No config loaded".to_string()))?;
+
+        info!(
+            "Testing database {} on server: {}",
+            config.database.database_name.to_string(),
+            config.database.host.to_string()
+        );
+
+        let connection_manager = get_storage_connection_manager(&config.database);
+        let result = connection_manager.execute("select 1");
+
+        if result.is_ok() {
+            Ok("Successfully connected to database".to_string())
+        } else {
+            Err(anyhow!(
+                "Failed to connect to database: {:?}",
+                result.err().unwrap()
+            ))
+        }
+    }
+}
+
+struct LoginTest;
+
+#[async_trait]
+impl Test for LoginTest {
+    async fn run(&self, test_data: &mut TestData) -> Result<String> {
+        let config = test_data
+            .server_config
+            .as_ref()
+            .ok_or(anyhow!("No config loaded".to_string()))?;
+
+        let username = test_data
+            .args
+            .username
+            .clone()
+            .unwrap_or("admin".to_string());
+        let password = test_data
+            .args
+            .password
+            .clone()
+            .unwrap_or("pass".to_string());
```

Comment:
These defaults really only help us in development right? This is going to be a publicly available tool... would probably just make username and password required arguments?